src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs

```diff
@@ -467,6 +468,53 @@ public IActionResult HasTheApprenticeBeenMadeRedundant(MadeRedundantViewModel vi
         }
 
+        [Route("{apprenticeshipHashedId}/details/whystopapprenticeship")]
+        [HttpGet]
+        public async Task<IActionResult> WhyStopApprenticeship(WhyStopApprenticeshipRequest request)
+        {
+            var viewModel = await _modelMapper.Map<WhyStopApprenticeshipViewModel>(request);
+            return View(viewModel);
+        }
+
+        [Route("{apprenticeshipHashedId}/details/whystopapprenticeship",Name = RouteNames.WhyStopApprenticeship)]
+        [HttpPost]
+        public IActionResult WhyStopApprenticeship(WhyStopApprenticeshipViewModel viewModel)
+        {
+            switch (viewModel.SelectedStatusChange)
+            {
+                case StopStatusReason.LeftEmployment:
+                    return RedirectToAction(nameof(StopApprenticeship), new { viewModel.AccountHashedId, viewModel.ApprenticeshipHashedId });
+
+                case StopStatusReason.ChangeProvider:
+                    return RedirectToAction(nameof(StopApprenticeship), new { viewModel.AccountHashedId, viewModel.ApprenticeshipHashedId });
+
+                case StopStatusReason.Withdrawn:
+                    return RedirectToAction(nameof(StopApprenticeship), new { viewModel.AccountHashedId, viewModel.ApprenticeshipHashedId });
+
+                case StopStatusReason.NeverStarted:
+                    return RedirectToAction(nameof(ApprenticeshipNeverStarted), new { viewModel.AccountHashedId, viewModel.ApprenticeshipHashedId });
+
+                default:
+                    return RedirectToAction(nameof(ApprenticeshipDetails), new ApprenticeshipDetailsRequest { AccountHashedId = viewModel.AccountHashedId, ApprenticeshipHashedId = viewModel.ApprenticeshipHashedId });
+            }
+        }
+
+        [Route("{apprenticeshipHashedId}/details/apprenticeshipneverstarted")]
+        [HttpGet]
+        public async Task<IActionResult> ApprenticeshipNeverStarted(ApprenticeshipNeverStartedRequest request)
+        {
+            var viewModel = await _modelMapper.Map<ApprenticeshipNeverStartedViewModel>(request);
+            return View(viewModel);
+        }
+
+        [Route("{apprenticeshipHashedId}/details/apprenticeshipneverstarted", Name = RouteNames.ApprenticeshipNeverStarted)]
+        [HttpPost]
+        public IActionResult ApprenticeshipNeverStarted(ApprenticeshipNeverStartedViewModel viewModel)
+        {
+            //return RedirectToAction(nameof(HasTheApprenticeBeenMadeRedundant), new ApprenticeshipDetailsRequest { AccountHashedId = viewModel.AccountHashedId, ApprenticeshipHashedId = viewModel.ApprenticeshipHashedId });
+            return RedirectToAction(nameof(HasTheApprenticeBeenMadeRedundant), new { viewModel.AccountHashedId, viewModel.ApprenticeshipHashedId, viewModel.IsCoPJourney, viewModel.StopMonth, viewModel.StopYear });

```

Comment:
this is still not working. StopMonth and StopYear are not included in the POST

Reply:
Added StopMonth and StopYear